common/src/View/MapDocument.cpp

```diff
@@ -2039,65 +2039,53 @@ namespace TrenchBroom {
                 ));
             }
 
-            const bool lockTexturesPref = pref(Preferences::TextureLock);
-            auto errorMutex = std::mutex{};
-            bool transformFailed = false;
-            auto transformErrors = std::vector<std::string>{};
-            auto transformResult = std::vector<std::optional<
-                std::pair<Model::Node*, Model::NodeContents>>>{nodesToTransform.size(), std::nullopt};
-
-            kdl::parallel_for(nodesToTransform.size(), [&](const size_t i) {
-                Model::Node* node = nodesToTransform[i];
+            using TransformResult = kdl::result<std::pair<Model::Node*, Model::NodeContents>, Model::BrushError, std::string>;
```

Comment:
I don't think we need the `std::string` here

Reply:
Ah, it's for `return "Unexpected world node";` etc. I get it now, but it does make code hard to understand initially when there's 2 chained implicit conversions (`char[]` -> `std::string` -> `TransformResult`)

Reply:
I'm also fine with just `ensure`ing that there is no world or layer node in the input data. That would be a bug anyway. Then we don't need the strings at all.

Reply:
See 526f8ff9a.